### ISSUE TITLE
Update kustomize templates

### ISSUE THREAD

**Issue by ArangoGutierrez** [kubernetes-sigs/node-feature-discovery#1241]
**What would you like to be added**:
When running `make deploy`  with a latest (devel) version

```bash
[eduardo@fedora node-feature-discovery]$ kustomize version
(devel)
```
A warning of deprecation is showing for NFD templates 

**Why is this needed**:
```bash
kubectl apply -k .
# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
```

**Comment by fmuyassarov** [comment]
Yes, we basically need to move for example - ../../base/rbac under https://github.com/kubernetes-sigs/node-feature-discovery/blob/febebd4afd7f27e2ed2eb17a32df1cd68c41eb92/deployment/overlays/prune/kustomization.yaml#L6 to be under https://github.com/kubernetes-sigs/node-feature-discovery/blob/febebd4afd7f27e2ed2eb17a32df1cd68c41eb92/deployment/overlays/prune/kustomization.yaml#L9. And same changes on the other kustomization files.

**Comment by marquiz** [comment]
Yeah, should be trivial to fix
/help

**Comment by ArangoGutierrez** [comment]
@dipankardas011 sure! we would help you along the way with reviews, go ahead! thanks for contributing!

**Comment by marquiz** [comment]
> can I try to fix it?

@dipankardas011 yeah, sure

**Comment by dipankardas011** [comment]
Created the PR!
